### PR TITLE
MM-35681 Detox/E2E: Add e2e for bot account, mention badges, at mention

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added test for MM-T511, MM-T520, MM-T1824 in
    - `detox/e2e/test/autocomplete/at_mention.e2e.js`
    - `detox/e2e/test/bot_account/bot_account.e2e.js`
    - `detox/e2e/test/notifications/mention_badges.e2e.js`
- Added new server api in
    - `detox/e2e/support/server_api/bot.js`
    - `detox/e2e/support/server_api/user.js`
- Fixed param comments and fixed/added profile picture helper functions

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-35681

Zephyr cases:
- [MM-T511](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T511)
- [MM-T520](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T520)
- [MM-T1824](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T1824)

#### Screenshots
Screen Shot 2021-07-12 at 12 46 56 PM
Screen Shot 2021-07-12 at 3 35 14 PM
Screen Shot 2021-07-12 at 4 10 13 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: detox/e2e/support/ui/screen/user_profile.js [hunk 1]
```diff
@@ -11,13 +11,12 @@ class UserProfileScreen {
...
         userProfileScreen: 'user_profile.screen',
         customStatus: 'user_profile.custom_status',
-        profilePicture: 'user_profile.profile_picture',
         userProfileScrollView: 'user_profile.scroll_view',
         closeSettingsButton: 'close.settings.button',
         editButton: 'user_profile.edit.button',
         userProfileBotTag: 'user_profile.bot_tag',
         userProfileDisplayName: 'user_profile.display_name',
-        userProfileGuestTag: 'user_profile.bot_tag',
+        userProfileGuestTag: 'user_profile.guest_tag',
```
Comment (saturninoabril) [inline detox/e2e/support/ui/screen/user_profile.js:19]: Good catch!

File: detox/e2e/test/notifications/mention_badges.e2e.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        await goToChannel(testPrivateChannel.display_name);
+        await openTeamSidebar();
+        await getTeamByDisplayName(testTeam1.display_name).tap();
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    const expectMentionBadges = async (messagePostedChannel) => {
```
Comment (saturninoabril) [inline detox/e2e/test/notifications/mention_badges.e2e.js:74]: nit: Would be nice to see the two `it` blocks first and move this function below.

**Review by saturninoabril — APPROVED** [review]
Thanks @josephbaylon! Tested and passed. LGTM, except for non-blcoking suggestion.

File: detox/e2e/test/notifications/mention_badges.e2e.js (hunk 2 above)
Comment (josephbaylon) [reply detox/e2e/test/notifications/mention_badges.e2e.js:82]: Updated.

**Review by enahum — APPROVED** [review]

**Pull request merged**